### PR TITLE
Search & Filter Events - TESTS (E2E)

### PR DESCRIPTION
**Pull request by tamaracvjetkovic**
Good evening, _colleague_! 🌚🌙  
I hope everything is well! 😊 

Here is the summary of what I did in this PR.

### `NOTE` 📝

> You will see that I used JDBC Template and raw database informations while updating the events' dates. I tried to @Autowire the services and repositories, but there was an underlying issue which I could not understand and resolve. Updating the rows directly with the JDBC Template was the only solution I found that worked correctly.

> Even though, unfortunately, our base's url, username and password are currently (very and visibly) compromised, I am glad that I found something that at least does the job. 

> If you find another way to do this (without sharing our vulnerable informations to everyone 😄), please inform me ASAP, so I can change the logic.

---

### 🟩 NEW:

> E2E BASE
- `chrome`
  - added a test **base** for **_Chrome_** **driver**
- `pages`
  - added a **HomePage** class consisting of the **web elements** I needed and used in testing
  - used **@FindBy** for more efficient and elegant element finding
 
> EVENTS (_TESTS_)
- `filters`:
  - tested all possible **combinations** of event _**filters**_ ✅ 
- `search`:
  - tested a few examples of **searching** by event **_name_** ✔️  
  - tested a few examples of **searching** by event **_description_** ✔️ 
- `combination`
  - tested **combinations** of **_filters_** and **_search_** ✅ 

---

That's it for now.
Please review my code and feel free to comment on anything you think needs to be changed, added or deleted.

> Thank you for your time. 🌞
> </> Best regards and happy coding! </>
> 
> ~ Tamara 🦔

### CODE REVIEW

File: Eventy/src/test/java/e2e/tests/SearchAndFilterEventsTest.java [hunk 1]
```diff
@@ -0,0 +1,1467 @@
...
+        slowDown();
+
+        List<WebElement> eventCards = home.getFilteredEvents();
+        assertEquals(eventCards.size(), 0);
+
+        assertEquals(driver.findElement(By.cssSelector(".mat-mdc-paginator-range-label")).getText().strip(), "0 of 0");
+    }
+
+    @Test
+    public void filterEvents_MaxParticipantsInvalid_DoesNotFilter() {
```
Comment (Rogan003) [inline Eventy/src/test/java/e2e/tests/SearchAndFilterEventsTest.java:131]: Im not experienced enough with testing to recommend some good practices, but would it be better in your opinion to name this in a way that suggests that this is an edge case test for number 0? Also is there a test for a negative number (if it can even be placed ofc)?

**Review by Rogan003 — APPROVED** [review]
Hi colleague 🦔  🎀 !

Wow, what a huge PR! 

Honestly it is even a bit too much for me to look at with true attention! In my opinion this is well, you structured it nicely and tested many combinations of possible scenarios. 

I would only suggest for you to think about edge cases and see if you can add something (I am not really all that deep in this to be able to think about it well). My thinking is this, for every scenario test a known valid case, edge cases and a known invalid case. Example (for start and end date): you did well by testing start date, end date and both. I would maybe add some edge case like start date  = end date, maybe event end date < start date for a clearly invalid case.

Anyway, you are much more experienced than me! You definitely know better, I am just saying something that is on top of my head, in case I can inspire you to do even better. With that being said, of course I approve this! If you want, you can also include Matijas in this, maybe he can find some nice tests that my eyes didn't see. And if you think about what I said and don't find any place with some specific edge case missed, then you don't have to change anything since this looks perfect!

Great job and approved 🏆 🦔  🎀
